ngraph/core/include/ngraph/op/util/nms_base.hpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "ngraph/op/op.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace util
+        {
+            /// \brief Base class for operations NmsBase and MatrixNms
+            ///
+            class NGRAPH_API NmsBase : public Op
+            {
+            public:
+                NGRAPH_RTTI_DECLARATION;
+                enum class SortResultType
+                {
+                    CLASSID, // sort selected boxes by class id (ascending) in each batch element
+                    SCORE,   // sort selected boxes by score (descending) in each batch element
+                    NONE     // do not guarantee the order in each batch element
+                };
+
+                NmsBase() = default;
+
+                /// \brief Constructs a NmsBase operation
+                ///
+                /// \param boxes Node producing the box coordinates
+                /// \param scores Node producing the box scores
+                /// \param sort_result Specifies order of output elements
+                /// \param output_type Specifies the output tensor type
+                /// \param iou_threshold Specifies intersection over union threshold
+                /// \param score_threshold Specifies minimum score to consider box for the
+                /// processing
+                /// \param nms_top_k Specifies maximum number of boxes to be selected per
+                /// class, -1 meaning to keep all boxes
+                /// \param keep_top_k Specifies maximum number of boxes to be selected per
+                /// batch element, -1 meaning to keep all boxes
+                /// \param background_class Specifies the background class id, -1 meaning to keep
+                /// all classes
+                /// \param nms_eta Specifies eta parameter for adpative NMS, in close range [0, 1.0]
+                NmsBase(const Output<Node>& boxes,
```

Comment:
Looks like you don't need to have optional arguments, because you always initialise allarguments.

Reply:
Done